lib/core/test/unit/Cardano/WalletSpec.hs

```diff
@@ -488,6 +496,70 @@ walletListTransactionsSorted wallet@(wid, _, _) _order (_mstart, _mend) history
                 (\(tx, meta) -> (txId tx, slotIdTime (meta ^. #slotId))) <$> history
         times `shouldBe` expTimes
 
+
+{-------------------------------------------------------------------------------
+                        Properties of tx fee estimation
+-------------------------------------------------------------------------------}
+
+-- | Properties of 'estimateFeeForCoinSelection':
+-- 1. There is no coin selection with a fee above the estimated maximum.
+-- 2. The minimum estimated fee is no greater than the maximum estimated fee.
+-- 3. Around 10% of fees are below the estimated minimum.
+prop_estimateFee :: NonEmptyList (Either String FeeGen) -> Property
+prop_estimateFee (NonEmpty results) = case actual of
+    Left err -> label "errors: all" $
+        Left err === head results
+    Right (W.FeeEstimation minFee maxFee) ->
+        label ("errors: " <> if any isLeft results then "some" else "none") $
+        counterexample ("minFee=" <> show minFee <> " maxFee=" <> show maxFee) $
```

Comment:
```suggestion
    Right estimation@(W.FeeEstimation minFee maxFee) ->
        label ("errors: " <> if any isLeft results then "some" else "none") $
        counterexample (show estimation) $
```

(Since `W.FeeEstimation` already has a `Show` instance.)